frontend/packages/console-dynamic-plugin-sdk/scripts/generate-doc.ts

```diff
@@ -41,3 +45,182 @@ renderTemplate('scripts/templates/console-extensions.md.ejs', {
   printComments: (docComments: string[]) => printJSDocComments(docComments).replace(/\n/g, '<br/>'),
   escapeTableCell: (value: string) => value.replace(/\|/g, '\\|'),
 });
+
+type ComponentInfo = {
+  name: string;
+  doc: any;
+};
+
+const EXAMPLE = '@example';
+const DYNAMIC_PKG_PATH = '@console/dynamic-plugin-sdk/';
+
+const renderDocNode = (docNode: tsdoc.DocNode): string => {
+  let result: string = '';
+  if (docNode) {
+    // Todo (bipuladh): Improve support for links.
+    if (docNode instanceof tsdoc.DocExcerpt) {
+      result += docNode.content.toString();
+    }
+    for (const childNode of docNode.getChildNodes()) {
+      result += renderDocNode(childNode);
+    }
+  }
+  return result;
+};
+
+const generateGitLinkDoc = (filePath: string) => ({
+  summary: `[For more details please refer the implementation](${filePath})`,
+});
+
+const generateDoc = (comment: tsdoc.DocComment) => {
+  const summary = renderDocNode(comment.summarySection);
+  const exampleBlock = comment.customBlocks.find((block) => block?.blockTag?.tagName === EXAMPLE);
+  const example = renderDocNode(exampleBlock?.content);
+  const parameters = comment.params.blocks.map((param) => {
+    let { parameterName } = param;
+    let description = renderDocNode(param.content);
+    if (!parameterName) {
+      parameterName = description.substring(1, description.indexOf('-') - 1);
+      description = description.substring(description.indexOf('-') + 1);
+    }
+    return {
+      parameterName,
+      description,
+    };
+  });
+  const returns = renderDocNode(comment.returnsBlock?.content);
+  return {
+    summary,
+    example,
+    parameters,
+    returns,
+  };
+};
+
+const getDocPath = (relPath: string, absolutePath?: string): string => {
+  // Uses '@console/dynamic-plugin-sdk' based imports
+  if (!absolutePath) {
+    const slicedPath = relPath.replace(DYNAMIC_PKG_PATH, '');
+    return resolvePath(slicedPath);
+  }
+  const dirName = path.dirname(absolutePath);
+  return require.resolve(path.resolve(dirName, relPath));
+};
+
+const consoleLinkGenerator = (fileLocation: string, isRequire: boolean) => {
+  const WORKSPACE = '@';
+  const ABSOLUTE = '/frontend/';
+  const INTERNAL_PKG = '@console/internal';
+  const WEB_ROOT_LINK = 'https://github.com/openshift/console/tree/master/frontend';
```

Comment:
I don't know if it matters right now -- but this will never work for older commits / branches. Master can get out of sync fast.

This will do for now I think, just so we generate something, but we should think on this some more.

@spadgett This will get us a link to the `master` file, but only at time of generation. If we move a file around or restructure in anyway, this link will break.

Reply:
Would it work to use `git rev-parse HEAD` for the SHA and use that for the link?

Reply:
Should we create a env var that is taken during compilation and we can make it point to specific branches?

Reply:
Maybe just a node [execSync](https://nodejs.org/api/child_process.html#child_processexecsynccommand-options) call to get the value while it's running? 🤔 

I'm not a Node expert by any means, so perhaps this won't work.